### ISSUE TITLE
Missing temperature on last 2 cores (rk3588)

### ISSUE THREAD

**Issue by System64fumo** [htop-dev/htop#1530]
Hello, As of a few hours ago [this](https://github.com/htop-dev/htop/issues/1160) issue has been solved and temperature finally shows up on rk3588. *Woo!*

However temperature is not shown for the last 2 cores.
![screenshot](https://github.com/user-attachments/assets/3c158888-02b7-497a-aa8e-b1f6c35dfba6)

Context:
Orange Pi 5 plus 16gb
Armtix linux
Linux 6.11 rc1 (Mainline)

**Comment by BenBE** [comment]
Can you check if the code block added by the patch in #1411 is ran 3 times per temperature update cycle: Once for each of the sub-blocks?

**Comment by System64fumo** [comment]
Nope
Seems like they run twice
From what i can see via `sensors -u` and [mainline's device tree](https://github.com/torvalds/linux/blob/20371ba120635d9ab7fc7670497105af8f33eb08/arch/arm64/boot/dts/rockchip/rk3588-base.dtsi#L2381) they do not have bigcore1_thermal, But have bigcore2_thermal instead.

I'm confused as to why that is the case?
Either way changing bigcore1_thermal to bigcore2_thermal in all instances solves this.

**Comment by BenBE** [comment]
Any objections to the following patch proposal?

```diff
diff --git a/linux/LibSensors.c b/linux/LibSensors.c
index 8ac34980..7d6d17e9 100644
--- a/linux/LibSensors.c
+++ b/linux/LibSensors.c
@@ -233,7 +233,7 @@ void LibSensors_getCPUTemperatures(CPUData* cpus, unsigned int existingCPUs, uns
                coreTempCount += 2;
                continue;
             }
-            if (String_eq(chip->prefix, "bigcore1_thermal")) {
+            if (String_eq(chip->prefix, "bigcore1_thermal") || String_eq(chip->prefix, "bigcore2_thermal")) {
                data[7] = temp;
                data[8] = temp;
                coreTempCount += 2;
```

**Comment by System64fumo** [comment]
This would also need `bigcore2_thermal` to be added to line 145 to work but other than that looks good to me.

**Comment by BenBE** [comment]
Okay, updated proposal:

```diff
diff --git a/linux/LibSensors.c b/linux/LibSensors.c
index 8ac34980..1b1753d3 100644
--- a/linux/LibSensors.c
+++ b/linux/LibSensors.c
@@ -142,6 +142,7 @@ static int tempDriverPriority(const sensors_chip_name* chip) {
       { "littlecore_thermal", 0 },
       { "bigcore0_thermal",   0 },
       { "bigcore1_thermal",   0 },
+      { "bigcore2_thermal",   0 },
       /* Low priority drivers */
       { "acpitz",             1 },
    };
@@ -233,7 +234,7 @@ void LibSensors_getCPUTemperatures(CPUData* cpus, unsigned int existingCPUs, uns
                coreTempCount += 2;
                continue;
             }
-            if (String_eq(chip->prefix, "bigcore1_thermal")) {
+            if (String_eq(chip->prefix, "bigcore1_thermal") || String_eq(chip->prefix, "bigcore2_thermal")) {
                data[7] = temp;
                data[8] = temp;
                coreTempCount += 2;
```

**Comment by System64fumo** [comment]
LGTM :+1:

**Comment by BenBE** [comment]
Hopefully now fixed by 179182d59f9ffbb230d1641762e131fb43856ba9 …
